### PR TITLE
SAK-29895 Webservice to change the user locale.

### PR DESCRIPTION
**Pull request by mcarropellicer**
SAK-29895 Webservice to change the user locale.

### CODE REVIEW

**Comment by ottenhoff** [comment]
It seems like anyone can run this webservice......... is it supposed to be limited to super-users?

**Comment by mcarropellicer** [comment]
Should be limited to super-users? Does users have permissions to change other's locale using the PreferencesService?

**Comment by ern** [comment]
I would say if user is updating their own preference, or an admin user can update anyone's preference.

File: webservices/cxf/src/java/org/sakaiproject/webservices/SakaiScript.java [hunk 1]
```diff
@@ -398,6 +403,57 @@ public String changeUserPassword(
...
+
+        UserEdit userEdit = null;
+        PreferencesEdit prefs = null;
+        try {
+            User user = userDirectoryService.getUserByEid(eid);
+            
+            try {
+                prefs = (PreferencesEdit) preferencesService.edit(user.getId());
+            } catch (Exception e1) {
+                prefs = (PreferencesEdit) preferencesService.add(user.getId());
```
Comment (ern) [inline webservices/cxf/src/java/org/sakaiproject/webservices/SakaiScript.java]: This try shoud be looking for IdUnusedException for doing an add.

**Pull request merged**